libs/core/src/lib/status-indicator/status-indicator.component.ts

```diff
@@ -0,0 +1,346 @@
+        return this._elementRef;
+    }
+    calculateFilling(): number {
+        this.fillCalculator = (this.fillPercentage * this.path.length) / 100;
+        return this.fillCalculator;
+    }
+
+    angleCalculation(): void {
+        let sPointsAttributeValue = [];
+        let polygonPoints: any[];
+        if (this.fillingType === 'angled') {
+            this.binaryString = this.convertAngleToBinary(this.angle);
+            this.assignBinaryValue(this.binaryString);
+        } else if (this.fillingType === 'radial') {
+            const tempPercent = this.fillCalculator % 1;
+            const fillNumber = Number(tempPercent.toFixed(2));
+
+            const element = this._elementRef.nativeElement.querySelectorAll('svg');
+
+            for (let i = 1; i < element.length; i++) {
+                sPointsAttributeValue = this._getPolygonPointsForCircularFilling(
+                    fillNumber * 100,
+                    element[i].getBBox()
+                );
+                polygonPoints = sPointsAttributeValue.reduce((acc, item) => {
+                    return acc + item.x + ',' + item.y + ' ';
+                }, '');
+                this.pointsArray.push(polygonPoints);
+            }
+        } else if (this.fillingType === 'linearup') {
+            this.binaryString = this.convertAngleToBinary(90);
+            this.assignBinaryValue(this.binaryString);
+        } else if (this.fillingType === 'linearleft') {
+            this.binaryString = this.convertAngleToBinary(180);
+            this.assignBinaryValue(this.binaryString);
+        } else if (this.fillingType === 'lineardown') {
+            this.binaryString = this.convertAngleToBinary(270);
+            this.assignBinaryValue(this.binaryString);
+        }
+    }
+
+    convertAngleToBinary(angle: number): string {
+        if (angle > 0 && angle <= 45) {
+            return '1,0,0,1';
+        } else if (angle >= 45 && angle < 90) {
+            return '0,0,0,1';
+        } else if (angle >= 90 && angle < 135) {
+            return '0,0,0,1';
+        } else if (angle >= 135 && angle < 180) {
+            return '0,0,1,1';
+        } else if (angle >= 180 && angle < 225) {
+            return '0,0,1,0';
+        } else if (angle >= 225 && angle < 270) {
+            return '0,1,1,0';
+        } else if (angle >= 270 && angle < 315) {
+            return '0,1,0,0';
+        } else if (angle >= 315 && angle < 360) {
+            return '1,1,0,0';
+        } else if (angle === 0 || angle === 360) {
+            return '1,0,0,0';
+        } else {
+            return 'invalid';
+        }
+    }
+    assignBinaryValue(binaryString: string): void {
+        let binaryValue = [];
+        binaryValue = binaryString.split(',');
+        this.x1 = binaryValue[0];
+        this.y1 = binaryValue[1];
+        this.x2 = binaryValue[2];
+        this.y2 = binaryValue[3];
+    }
+
+    _createPoint(iX, iY): any {
+        return { x: iX, y: iY };
+    }
```

Comment:
You should not use `any`, instead define a Point interface and use that in place.